oidcprovider/oidc.go

```diff
@@ -0,0 +1,124 @@
+package oidcprovider
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/openpubkey/openpubkey/gq"
+	"github.com/openpubkey/openpubkey/pktoken"
+	"github.com/openpubkey/openpubkey/util"
+)
+
+const gqSecurityParameter = 256
+
+type PublicKey interface {
+	Equal(x crypto.PublicKey) bool
+}
+
+var ErrNonGQUnsupported = fmt.Errorf("non-GQ signatures are not supported")
+
+// Interface for interacting with the OP (OpenID Provider)
+type OpenIdProvider interface {
+	RequestTokens(ctx context.Context, cicHash string) ([]byte, error)
+	PublicKey(ctx context.Context, idt []byte) (PublicKey, error)
+	NonceClaimName() string
+	VerifyOIDCSig(ctx context.Context, idt []byte, expectedNonce string) error
```

Comment:
Why not just `VerifySignature`?

Reply:
It's to differentiate from the GQ signature - this is for verifying the signature from the OIDC provider where a GQ signature is not present.

To be honest, I'm thinking more and more that we should simply always use GQ signatures. We already made it an error not to use GQ signatures for Github Actions, maybe we should do the same for Google and future providers too... WDYT @EthanHeilman ?

Reply:
That makes a lot of sense, maybe there could be a boolean arg or filed in opts struct to skip.

Reply:
I think it would be nice to add a new claim to the `cic` for `zkp=gq` (or something) and not allowing it to be `none` so that we can always default to `gq` for now, and not have to maintain the vanilla oidc verification, but still allow for flexibility if there are any issues we find with `gq` down the line (security-wise, resource-wise, or other).

Reply:
@jonnystoten For the workload identity I think we should require GQ signatures. In the user identity, we want allow both GQ and non-GQ.:
* GQ signatures are large and would introduce noticeable latency for say a user logging into a host over SSH or signing an API request in a browser.
* In the access use case, PK Tokens are not public so GQ signatures are less needed.

Reply:
I created this issue to discuss the best way to flag that GQ signatures are being used. Please comment thoughts and ideas on that issue and then we can discuss and come to a conclusion.
https://github.com/openpubkey/openpubkey/issues/15